### PR TITLE
Added from and to parameters to top N queries to specify time range

### PR DESCRIPTION
**Pull request by LilyCaroline17**
### Description
Added from and to parameters to documentation for top N queries
References PR: [opensearch-project/query-insights#84](https://github.com/opensearch-project/query-insights/pull/84)
Issues PR: [opensearch-project/query-insights#12](https://github.com/opensearch-project/query-insights/issues/12)

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _observing-your-data/query-insights/top-n-queries.md [hunk 1]
```diff
@@ -96,6 +96,16 @@ GET /_insights/top_queries?type=memory
 ```
 {% include copy-curl.html %}
 
+To specify a time range for querying top N results, use the `from` and `to` parameters in ISO8601 format: `YYYY-MM-DD'T'HH:mm:ss.SSSZ`.
+For example, to retrieve the top N queries from August 25, 2024, at 15:00 UTC to August 30, 2024, at 17:00 UTC, make the following request:
```
Comment (kolchfa-aws) [inline _observing-your-data/query-insights/top-n-queries.md:100]:
```suggestion
For example, to retrieve the top N queries from August 25, 2024, at 15:00 UTC to August 30, 2024, at 17:00 UTC, send the following request: 
```

File: _observing-your-data/query-insights/top-n-queries.md [hunk 2]
```diff
@@ -96,6 +96,16 @@ GET /_insights/top_queries?type=memory
...
 
+To specify a time range for querying top N results, use the `from` and `to` parameters in ISO8601 format: `YYYY-MM-DD'T'HH:mm:ss.SSSZ`.
+For example, to retrieve the top N queries from August 25, 2024, at 15:00 UTC to August 30, 2024, at 17:00 UTC, make the following request: 
+
+```json
+GET /_insights/top_queries?from=2024-08-25T15:00:00.000Z&to=2024-08-30T17:00:00.000Z
+```
+{% include copy-curl.html %}
+
+If you have [local index exporter enabled](#configuring-a-local-index-exporter), historical queries stored in local OpenSearch indexes will also be included within the specified time range.
```
Comment (kolchfa-aws) [inline _observing-your-data/query-insights/top-n-queries.md:107]:
```suggestion
If you have a [local index exporter enabled](#configuring-a-local-index-exporter), historical queries stored in local OpenSearch indexes will also be included within the specified time range. 
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @LilyCaroline17! Just a couple of suggestions.

**Comment by kolchfa-aws** [comment]
I'm assuming that this is part of 2.17 so adding the 2.17 label.

**Comment by kolchfa-aws** [comment]
Thank you! Applying "Done but waiting to merge" label to put this in the 2.17 editorial queue starting 9/10. This PR will be merged before the release.

**Comment by ansjcy** [comment]
@kolchfa-aws Let's not merge this one in 2.17, since we still need some time to review the underlying PR: https://github.com/opensearch-project/query-insights/pull/84 and we might skip it for 2.17.

**Comment by kolchfa-aws** [comment]
@ansjcy Sure, I'll put it on hold for now.

**Comment by kkhatua** [comment]
@ansjcy  should we mark this for 2.18  if ew're not ready with the feature merge for 2.17 ?

File: _observing-your-data/query-insights/top-n-queries.md [hunk 3]
```diff
@@ -96,6 +96,16 @@ GET /_insights/top_queries?type=memory
...
 
+To specify a time range for querying top N results, use the `from` and `to` parameters in ISO8601 format: `YYYY-MM-DD'T'HH:mm:ss.SSSZ`.
+For example, to retrieve the top N queries from August 25, 2024, at 15:00 UTC to August 30, 2024, at 17:00 UTC, send the following request: 
+
+```json
+GET /_insights/top_queries?from=2024-08-25T15:00:00.000Z&to=2024-08-30T17:00:00.000Z
+```
+{% include copy-curl.html %}
+
+If you have a [local index exporter enabled](#configuring-a-local-index-exporter), historical queries stored in local OpenSearch indexes will also be included within the specified time range.
```
Comment (natebower) [inline _observing-your-data/query-insights/top-n-queries.md:107]:
```suggestion
If you have a [local index exporter enabled](#configuring-a-local-index-exporter), historical queries stored in local OpenSearch indexes will also be included in the specified time range. 
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @LilyCaroline17 Just one minor change. Thanks!

**Comment by kolchfa-aws** [comment]
@ansjcy Looks like the code is merged and backported to 2.x so this made it to 2.17. Removing the "on hold" label.

**Comment by ansjcy** [comment]
We are planing to ship this for 2.18 - as we merged the corresponding change into 2.x but didn't backport it into 2.17. Let's change the label to 2.18.

**Pull request merged**